assignment/impl/src/java/org/sakaiproject/assignment/impl/persistence/AssignmentRepositoryImpl.java

```diff
@@ -378,4 +387,102 @@ private void hardDeleteHelper(String assignmentId){
         }
 
     }
+
+    @Override
+    public List<SimpleAssignmentAutoSubmit> findAutoSubmitAssignmentsBySite(String siteId, java.time.Instant now) {
+        List<Assignment> assignments = startCriteriaQuery()
+                .add(Restrictions.eq("context", siteId))
+                .add(Restrictions.eq("deleted", Boolean.FALSE))
+                .add(Restrictions.eq("draft", Boolean.FALSE))
+                .add(Restrictions.le("closeDate", now))
+                .list();
+        return assignments.stream()
+                .filter(a -> {
+                    String autoSubmit = a.getProperties().get(AssignmentConstants.ASSIGNMENT_AUTO_SUBMIT_ENABLED);
+                    return "true".equals(autoSubmit);
+                })
+                .map(a -> {
+                    SimpleAssignmentAutoSubmit dto = new SimpleAssignmentAutoSubmit();
+                    dto.id = a.getId();
+                    dto.title = a.getTitle();
+                    dto.dueTime = a.getDueDate();
+                    dto.closeTime = a.getCloseDate();
+                    dto.context = a.getContext();
+                    dto.draft = a.getDraft();
+                    dto.isGroup = Boolean.TRUE.equals(a.getIsGroup());
+                    dto.properties = a.getProperties();
+                    return dto;
+                })
+                .collect(Collectors.toList());
+    }
```

Comment:
The repository should be about interacting with the DB only, this data object should be in the service.